# Crossmatch habitable zone targets with the CKS sample and my own rotation period catalogue.

In [1]:
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import pandas as pd

Load Stephen Kane's list of targets, the CKS catalogue, the McQuillan catalogue and my own rotation period catalogue.

In [14]:
# Kane catalogue:
kane = pd.read_csv("kane_targets.csv")

# CKS catalogue:
cks = pd.read_csv("cks_physical_merged.csv")

# McQuillan catalogue:
mc = pd.read_csv("Table_1_Periodic.txt")

# Tim and my catalogue:
tdmra = pd.read_csv("koi_periods_0712.csv")

What's the overlap between Stephen Kane's catalogue and the CKS sample?

In [44]:
# Format the CKS catalogue and Kane catalogue column headers.
koi_ids = []
for i in cks.id_koicand:
    koi_ids.append("Kepler-{}".format(int(float(i[1:]))))
cks["koi_id"] = koi_ids

KOIs = np.zeros(np.shape(kane)[0], dtype=int)
for j, i in enumerate(kane.koi_id):
    if i[:7] == "Kepler-":
        KOIs[j] = int((float(i[7:])))

kane["KOI"] = KOIs
print(kane.keys(), "\n")
print(cks.keys())
kane_cks = pd.merge(kane, cks, on="koi_id", how="inner")
print(np.shape(kane), np.shape(cks), np.shape(kane_cks))

Index(['koi_id', 'planet_designation', 'p_orb_days', 'a_AU', 'r_pl_earth',
       'teff', 'r_star_rsol', 'insolation_flux_F_E', 'KOI'],
      dtype='object') 

Index(['id_starname', 'id_kic', 'id_koicand', 'id_kepler_name',
       'koi_disposition', 'koi_period', 'koi_period_err1', 'koi_period_err2',
       'koi_time0', 'koi_time0_err1',
       ...
       'iso_sma', 'iso_sma_err1', 'iso_sma_err2', 'iso_insol',
       'iso_insol_err1', 'iso_insol_err2', 'iso_teq', 'iso_teq_err1',
       'iso_teq_err2', 'koi_id'],
      dtype='object', length=121)
(1162, 9) (2025, 121) (617, 129)


In [35]:
print(i)

K2-5


How many of these stars have rotation periods measured by Amy?

In [59]:
print(np.sort(kane_cks.id_kic.values)[:10])
print(np.sort(mc.kepid.values[:200]))
kane_cks_mc = pd.merge(kane_cks, mc, left_on="id_kic", right_on="kepid", how="inner")
np.shape(kane_cks_mc)

[ 757450 1865042 2162635 2306756 2438264 2438264 2446113 2446113 2571238
 2692377]
[ 892376 1026146 1026474 1162635 1164102 1292688 1297303 1429921 1430349
 1430893 1431060 1431116 1432700 1432745 1435229 1569682 1569863 1570915
 1571364 1572802 1573478 1577265 1718071 1721911 1722506 1849590 1865663
 1866535 1866924 1866986 1867358 1868579 1870952 1871120 1872885 1873523
 1873543 1996574 2010738 2012731 2013627 2013754 2014203 2015703 2015767
 2016979 2017692 2018383 2020185 2020638 2140909 2142372 2142451 2156061
 2156718 2157356 2160105 2161569 2162907 2163101 2163610 2163634 2167693
 2282712 2282944 2285598 2300039 2300440 2301872 2302851 2302997 2305760
 2309152 2311842 2423549 2424688 2435893 2436635 2437270 2437317 2438151
 2441409 2441562 2441891 2441989 2442004 2442866 2443037 2443364 2443935
 2444406 2445782 2446934 2448311 2448958 2554730 2555523 2557350 2557669
 2558314 2568503 2570277 2570846 2572352 2574013 2574427 2575433 2576013
 2576328 2577210 2577743 2581298 2581444 

(0, 140)

None... How about in Tim and my catalogue?

In [64]:
kane_cks_tdmra = pd.merge(kane_cks, tdmra, on="KOI", how="inner")
np.shape(kane_cks_tdmra)

(552, 142)

So there are 552 stars with spectroscopic properties AND rotation periods. The next thing to do is to combine all this information using isochrones and a gyrochronology model. See the ipython notebook called age_inference.ip